### ISSUE TITLE
Wrong RawType on enums

### ISSUE THREAD

**Issue by txaiwieser** [krzysztofzablocki/Sourcery#85]
I came across this today, don't know if its a bug, bug it was not what I was expecting to be the RawType

```swift
{% for enum in types.enums %}{% if enum.rawType %}
RawType => {{ enum.rawType }}
{% endif %}{% endfor %}


enum Test1 {
	case one
}
// Does not generate code // ok!



enum Test2: String {
	case two
}
// Does generate code
RawType => String // ok!



enum Test3: Int {
	case three
}
// Does generate code
RawType => Int // ok!



enum Test4: Equatable {
	case four
}
// Does generate code
RawType => Equatable // it this suppouse to be the RawType? I was expecting to be nil

```

Thanks 👍

**Comment by krzysztofzablocki** [comment]
I think it happens because we don't know what Equatable is, since it's not part of your project. /cc @ilyapuchka

**Comment by ilyapuchka** [comment]
Yes, that's the reason, but I will look at it again trying to improve the behaviour, so that it could be worked around by adding `Equatable` conformance in extension. Currently only using dummy protocol that extends `Equatable` will work, but extension approach sounds better.

**Comment by krzysztofzablocki** [comment]
maybe we should whitelist primitive types that can become RawType automatically? I think any complex type would require the protocol conformance that we are already looking for automaticaly?

**Comment by ilyapuchka** [comment]
We can, but that will be a pretty long list =)

**Comment by krzysztofzablocki** [comment]
@txaiwieser you will be able to apply that workaround @ilyapuchka implemented in 0.5.0, we'll do release once my PR is merged
